react_frontend/src/pages/games/memoryVault/MemoryVaultRecall.js

```diff
@@ -0,0 +1,222 @@
+      Recalled: recalledWords,
+      HintsUsed: hintsUsed,
+      Timestamp: new Date().toISOString(),
+    };
+
+    const formattedDate = new Date().toLocaleDateString('en-GB').replace(/\//g, '-');
+    const pathToDailyReports = `/users/${userId}/dailyReports/${formattedDate}/memoryVault/recallSpeedAndAccuracy`;
+    const pathToSeeMore = `/users/${userId}/dailyReportsSeeMore/${formattedDate}/memoryVault/recallSpeedAndAccuracy`;
+
+    try {
+      await setDoc(doc(db, pathToDailyReports), userAttempt);
+      await setDoc(doc(db, pathToSeeMore), userAttempt);
+    } catch (error) {
+      console.error("Error saving response:", error);
+    }
+
+    const newCount = playCount + 1;
+    setPlayCount(newCount);
+
+    const userRef = doc(db, "users", userId);
+    const userDoc = await getDoc(userRef);
+
+    if (userDoc.exists()) {
+      // today's date as a string adjusted to local time zone
+      const today = new Date();
+      today.setMinutes(today.getMinutes() - today.getTimezoneOffset());
+
+      // today's date in YYYY-MM-DD format
+      const formattedToday = today.toISOString().split('T')[0];
+
+      const yesterday = new Date();
+      yesterday.setDate(yesterday.getDate() - 1);
+      yesterday.setMinutes(yesterday.getMinutes() - yesterday.getTimezoneOffset());
+      const yesterdayStr = yesterday.toISOString().split("T")[0]; 
+
+      const lastPlayed = userDoc.data().lastPlayed || null;
+      const firstPlayed = userDoc.data().firstPlayed || formattedToday;
+
+      const completedDays = userDoc.data().completedDays || [];
+
+      console.log('completedDays', completedDays);
+      console.log('completedDays.length', completedDays.length);
+
+
+      let newStreak = 1;
+
+      if (lastPlayed) {
+        if (lastPlayed === yesterdayStr) {
+          newStreak = userDoc.currentStreak + 1;
+        }
+      }
+
+      await updateDoc(userRef, {
+        completedDays: arrayUnion(formattedToday),
+        numCompletedDays: completedDays.includes(formattedToday) ? completedDays.length : completedDays.length + 1,
+        playCount: newCount,
+        lastPlayed: formattedToday,
+        firstPlayed: firstPlayed,
+        currentStreak: newStreak
+      });
+    }
+
+    setInputWord("");
+    setInputAudio("");
+    setInputPicture("");
+    setHintsUsed(0);
+    setWordHint(null);
+    setAudioHint(null);
+    setPictureHint(null);
+  };
+
+  return (
+    <div className="memory-vault-container">
+      <div className="title-container-instructions">
+        <img src={titleImage} alt="Title" className="title-image-small" />
+      </div>
+
+      <div className="memory-vault-content">
+        <h1 className="game-title">Unlock the Memory Vault</h1>
+        <p className="game-instructions-text">Recall the items presented earlier in this session:</p>
+        <div className="memory-input-container">
+          <div className='input-container'>
+            <p className='bold'>Enter the word you saw</p>
```

Comment:
I think the wording is confusing because when I first when to play the game I thought this was referring to the picture because it says "saw"

```suggestion
            <p className='bold'>Enter the word you read</p>
```